backend/src/apiserver/resource/resource_manager.go

```diff
@@ -599,6 +607,9 @@ func (r *ResourceManager) ReportWorkflowResource(workflow *util.Workflow) error
 	}
 	runId := workflow.ObjectMeta.Labels[util.LabelKeyWorkflowRunId]
 	jobId := workflow.ScheduledWorkflowUUIDAsStringOrEmpty()
+	if len(workflow.Namespace) == 0 {
+		return util.NewInvalidInputError("Workflow missing namespace")
```

Comment:
Why is this needed? 
Are we sure this would work for "default" Kubernetes namespace?

Reply:
This is not needed. But I think it is clearer, because I think persistence agent is the only instance calling this. And it gets data using workflow client, so it will always carry namespace.

Reply:
I verified retry works. That includes coverage for persistence agent. Also, e2e tests pass.